sdk/ai/azure-ai-agents/samples/agents_tools/sample_agents_mcp.py

```diff
@@ -0,0 +1,169 @@
+# pylint: disable=line-too-long,useless-suppression
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with the
+    Model Context Protocol (MCP) tool from the Azure Agents service using a synchronous client.
+    To learn more about Model Context Protocol, visit https://modelcontextprotocol.io/
+
+USAGE:
+    python sample_agents_mcp.py
+
+    Before running the sample:
+
+    pip install azure-ai-projects azure-ai-agents azure-identity
+
+    Set these environment variables with your own values:
+    1) PROJECT_ENDPOINT - The Azure AI Project endpoint, as found in the Overview
+                          page of your Azure AI Foundry portal.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the AI model, as found under the "Name" column in
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+    3) MCP_SERVER_URL - The URL of your MCP server endpoint.
+    4) MCP_SERVER_LABEL - A label for your MCP server.
+"""
+
+import os, time
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.agents.models import McpTool, RequiredMcpToolCall, SubmitToolApprovalAction, ToolApproval
+
+# Get MCP server configuration from environment variables
+mcp_server_url = os.environ.get("MCP_SERVER_URL", "https://gitmcp.io/Azure/azure-rest-api-specs")
+mcp_server_label = os.environ.get("MCP_SERVER_LABEL", "github")
+
+project_client = AIProjectClient(
+    endpoint=os.environ["PROJECT_ENDPOINT"],
+    credential=DefaultAzureCredential(),
+)
+
+# [START create_agent_with_mcp]
+# Initialize agent MCP tool
+mcp_tool = McpTool(
```

Comment:
McpTool can be mixed with the tools that MCP provides.

How about `McpConnectorTool`?

Reply:
We shall check with architects, e.g. Johan